chainerx_cc/chainerx/routines/connection.cc

```diff
@@ -362,8 +362,9 @@ Array Linear(const Array& x, const Array& w, const absl::optional<Array>& b, uin
         }
     }
 
-    BackwardBuilder bb = has_bias ? BackwardBuilder{"linear", {x_matrix, w, b_matrix}, out_matrix}
-                                  : BackwardBuilder{"linear_nobias", {x_matrix, w}, out_matrix};
+    BackwardBuilder bb{has_bias ? "linear" : "linear_nobias",
+                       has_bias ? std::vector<ConstArrayRef>{x_matrix, w, b_matrix} : std::vector<ConstArrayRef>{x_matrix, w},
+                       out_matrix};
```

Comment:
Hmm, this does not look very elegant...
Is there any other way to fix the error?

Reply:
https://github.com/chainer/chainer/blob/f653368d7f296878360f3d98ff1c128fb2a51e1d/chainerx_cc/chainerx/backward_builder.h#L141
make it `= default;` Will also fix this, granted.

Reply:
OK, I will keep this for now. I tried `BackwardBuilder& operator=(BackwardBuilder&&) = default;` But I still need to std::move here...😂

Reply:
I think it's still better even if `std::move` is required. In the current code a single conditional is scattered into multiple expressions, which does not look elegant.
How do you think?

Reply:
Default move assignment is not needed anyway. Just move here will be OK. But default move constructor with `noexcept` makes MSVC unhappy.

Reply:
OK, please fix as such.
As for `noexcept`, it is required to suppress clang-tidy error.
Please feel free to report an issue though.